src/Application.php

```diff
@@ -91,9 +89,7 @@ public function addOptions(InputOption $options): void
 
     public function extractNamespace(string $name, int $limit = null): string
     {
-        $parts = explode('/', $name, -1);
-
-        return implode('/', null === $limit ? $parts : array_slice($parts, 0, $limit));
+        return parent::extractNamespace(str_replace('/', ':', $name), $limit);
     }
 
     public function getNamespaces(): array
```

Comment:
Here need to take into account that both `:` and `/`  can be used as a separator.